### ISSUE TITLE
Can‘t reproduce the experiment results which have large margins compared to the results in paper.

### ISSUE THREAD

**Issue by Beltakoll** [kinredon/Harmonious-Teacher#10]
Hi kinreon~ Thank you for your excellent work in CDOD. I have read your paper and found that the method really attracted me. When our team members reproduced the cross-domain tasks mentioned in your paper, the experiment results were not good. The details are as follows.
We use two single-GPU machines to conduct two tasks, C->F and S->C. We follow the readme to construct cityscapes and foggy cityscapes datasets. And We use the default config.py to train the model without any modification. The results are shown in the following table. 
![image](https://github.com/kinredon/Harmonious-Teacher/assets/157275213/49ccca25-85e2-4dd8-9f02-d40f939eef80)
and the AP50 curves when conducting C->F using RTX4090 is as follows:
![image](https://github.com/kinredon/Harmonious-Teacher/assets/157275213/9d736a1c-7bf1-4101-be82-25549b23bd79)
S->C also using RTX4090:
![image](https://github.com/kinredon/Harmonious-Teacher/assets/157275213/99742116-99f5-4152-bd3e-686663c61475)
As an example, the training commands are as follows:
`CUDA_VISIBLE_DEVICES=0`

`nohup pods_train --dir . --dist-url "tcp://127.0.0.1:29007" --num-gpus 1 OUTPUT_DIR 'outputs/ht_s2c' > log.txt 2>&1 &`

The experiment results are not good. Could you please give me some advice to figure out such circumstances?

Thank you a lot!~

**Comment by kinredon** [comment]
Hi, @Beltakoll. The default training schedule is designed for 4 GPUs, thus I suggest you use 4 RTX 4090/3090 GPUs to train the model. Other configurations may need to be further tuned for some hyperparameters.

**Comment by Simencassiman** [comment]
Hi, @Beltakoll have you managed to make any progress on this issue?
I'm running into the same problem with 4 GPUs, only achieving a best_map50.AP50 of 44.4, which is still quite a long way off from the paper results.
The code is exactly as it is in the repo except for the paths to load the annotations, which have been converted with the script provided in the repo. So I really don't see where the problem could lie.
The only difference is that I'm running on 4 x Tesla P100, but I don't believe this should have such an impact if all the configurations are the same.
@kinredon do you have any suggestions? Would it be possible for you to provide the trained weights?

**Comment by kinredon** [comment]
Hi @Simencassiman , can you provide your training log for analyzing the issue? I will try to find the trained weights and share them with you.

**Comment by Simencassiman** [comment]
Hi @kinredon 
Thank you very much for the fast reply!
Attached I provide the log file saved to wandb which seems to contain the relevant info.
[output.log](https://github.com/kinredon/Harmonious-Teacher/files/14613672/output.log)

**Comment by kinredon** [comment]
Hi @Simencassiman, It seems the number of samples in the validation set is not right. 

For example, in your log file, the inference samples is 13:
```
2024-03-14 03:47:18 | INFO     | cvpods.evaluation.evaluator:112 - Start inference on ***13*** data samples
2024-03-14 03:47:37 | INFO     | cvpods.evaluation.evaluator:142 - Inference done 11/13. 0.0859 s / sample. ETA=0:00:00
2024-03-14 03:47:38 | INFO     | cvpods.evaluation.evaluator:154 - Total inference time: 0:00:01.597664 (0.199708 s / sample per device, on 4 devices)
2024-03-14 03:47:38 | INFO     | cvpods.evaluation.evaluator:160 - Total inference pure compute time: 0:00:00 (0.085777 s / sample per device, on 4 devices)
2024-03-14 03:47:38 | INFO     | cvpods.evaluation.pascal_voc_evaluation:87 - Evaluating foggy_val using 2007 metric. Note that results do not use the official Matlab API.
2024-03-14 03:47:40 | INFO     | cvpods.evaluation.pascal_voc_evaluation:123 - Evaluation results for bbox:
|   AP   |  AP50  |  AP75  |
|:------:|:------:|:------:|
| 23.752 | 40.855 | 22.417 |
2024-03-14 03:47:40 | INFO     | cvpods.evaluation.pascal_voc_evaluation:125 - Evaluation results for bbox (per-category):
|  person  |  rider  |  car   |  truck  |  bus   |  train  |  motorcycle  |  bicycle  |
|:--------:|:-------:|:------:|:-------:|:------:|:-------:|:------------:|:---------:|
|  46.960  | 57.967  | 65.990 | 17.219  | 63.044 |  0.000  |    28.741    |  46.922   |
2024-03-14 03:47:40 | INFO     | cvpods.engine.runner:398 - Evaluation results for foggy_val in csv format:
2024-03-14 03:47:40 | INFO     | cvpods.evaluation.testing:26 - copypaste: Task: bbox
2024-03-14 03:47:40 | INFO     | cvpods.evaluation.testing:27 - copypaste: AP,AP50,AP75
2024-03-14 03:47:40 | INFO     | cvpods.evaluation.testing:28 - copypaste: 23.7524,40.8555,22.4171
```
You may check your validate set. The right number should be 125 (500 validation images for 4 GPUs).

**Comment by Simencassiman** [comment]
Hi @kinredon 
That does seem to have improved the situation! Silly mistake on my part...
Now, according to the included test_net script, the teacher AP50 is at 51.5 and the student is at 47.8. So is it the teacher metrics that you report? Or is there still something else going on?
Thank you very much for your help.

**Comment by kinredon** [comment]
@Simencassiman This is nice! I report the performance of the teacher model because it usually has higher performance than the student model.
